### PR TITLE
define data manager api for secondary storage 

### PR DESCRIPTION
**Pull request by Napoleon0621**
define secondary storage api
#149

### CODE REVIEW

File: storage/api/datamanager/datamanager.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package datamanager
+
```
Comment (git-hulk) [inline storage/api/datamanager/datamanager.go:2]: For how to organize those, I perfer below:

```
|__ storage
|______ storage.go(interface)
|______ protos(group all proto files here)
|______ spanner
|__________ Implement spanner here
|______ TiKV...
```

Comment (Napoleon0621) [reply storage/api/datamanager/datamanager.go:2]: Roger that, will do.

File: storage/api/datamanager/datamanager.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	// PumpOut pumps data which are due before certain due time
+	PumpOut(req []*DataRequest, limit int64) (jobs []*JobData, err error)
+	// GetStorageSize returns the size of data in storage which are due before certain due time
+	GetStorageSize(req []*DataRequest) (count int64, err error)
+	// DelByID remove job data from storage based on job id
+	DelByID(jobIDs []string) (count int64, err error)
+	// DelBeforeDue remove job which are due before certain due time from storage
+	DelBeforeDue(req []*DataRequest) (count int64, err error)
+	// DelBehindDue remove job which are due after certain due time from storage
+	DelBehindDue(req []*DataRequest) (count int64, err error)
```
Comment (git-hulk) [inline storage/api/datamanager/datamanager.go:16]:
```suggestion
	// AddJobs write jobs into the secondary storage 
    AddJobs(jobs []*JobData) (count int64, err error)
	// PumpOut pumps data which are due before certain due time
	BatchGetJobs(req []*DataRequest, count int64) (jobs []*JobData, err error)
	// GetQueueSize returns the size of data in storage which are due before certain due time
	GetQueueSize(req []*DataRequest) (count int64, err error)
	// DelJobs remove job data from storage based on job id
	DelJobs(jobIDs []string) (count int64, err error)
	// DelBeforeDue remove job which are due before certain due time from storage
	DelBeforeDue(req []*DataRequest) (count int64, err error)
	// DelBehindDue remove job which are due after certain due time from storage
	DelBehindDue(req []*DataRequest) (count int64, err error)
```
I'm not sure how can we use `DelBeforeDue` and `DelBehindDue`, can you explain a bit more?

Comment (Napoleon0621) [reply storage/api/datamanager/datamanager.go:16]: For now,  just ignore these functions.Will add them back in case of future need,.

Comment (git-hulk) [reply storage/api/datamanager/datamanager.go:16]: okay, let's remove them.

**Pull request closed without merging**